Telemetry improvements

vklimontovich opened the pull request:
## What does this PR do?

This PR improves telemetry for cal.com. It moves data collection to [next-collect](https://github.com/jitsucom/next-collect). It removes one 3rd-party call from the client-side rendering

**Environment**: Staging

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] Deploy latest version on Vercel and that middleware sends data to telemetry end-point. Set env `TELEMETRY_DEBUG=true` to see (in SSR console) what data points are collected

Flyover-ArtSk committed:
removed _middleware.ts for test deploy

Flyover-ArtSk committed:
naming changes

Flyover-ArtSk committed:
next-collect initial setup

vklimontovich committed:
WIP

vklimontovich committed:
WIP: added cookie name

Flyover-ArtSk committed:
telemetry update

Flyover-ArtSk committed:
Merge branch 'feat/next-collect' of https://github.com/jitsucom/cal.com into feat/next-collect

Flyover-ArtSk committed:
fixes

Flyover-ArtSk committed:
telemetry eventTypes fix

Flyover-ArtSk committed:
tmp jitsu config for testing

Flyover-ArtSk committed:
tmp jitsu key update (for tests)

Flyover-ArtSk committed:
deploy commit

Flyover-ArtSk committed:
cookieName fixes

Flyover-ArtSk committed:
telemetry credentials update

Flyover-ArtSk committed:
merge master

vklimontovich committed:
Merge remote-tracking branch 'origin/feat/next-collect' into main

# Conflicts:
#	apps/web/pages/_app.tsx

vklimontovich committed:
NextCollect updated to latest canary; added TELEMETRY_KEY to config

Flyover-ArtSk committed:
TELEMETRY_KEY fix

removed browser_user_agent field from event

Flyover-ArtSk committed:
_middleware.ts restored

Flyover-ArtSk committed:
next-collect version bump

Flyover-ArtSk committed:
yarn.lock fix

vklimontovich committed:
Merge branch 'calcom:main' into main

vklimontovich committed:
Added license consent property, set default telemetry endpoint

PeerRich reviewed (approved):
thank you a ton!

leog commented on apps/web/package.json:
```diff
@@ -81,6 +80,7 @@
     "mime-types": "^2.1.35",
     "next": "^12.1.6",
     "next-auth": "^4.3.4",
+    "next-collect": "canary",
```
Is this safe to use? it's using `canary` version corresponding to an alpha version 🤔

zomars replied:
He's the library author @leog so I assume he knows what he's doing 🤔

leog replied:
Oh I didn’t realize. To my point, shouldn’t we wait for a stable release to avoid issues @vklimontovich? I had very bad experiences when using unstable releases, but if we are OK with this, never mind me.

vklimontovich replied:
Yes, it safe. At the moment the latest canary is stable. We haven't figured out release policy yet. We'll do it in next few days, and I'll send an additional PR to switch to latest stream. Until then we'll not going to publish any canaries anyway

leog reviewed (changes requested):
LGTM, although I added a pending comment to agree upon before considering ready to merge

zomars commented on .env.example:
```diff
@@ -53,8 +53,8 @@ NEXTAUTH_SECRET=
 NEXTAUTH_COOKIE_DOMAIN=.example.com
 
-# Remove this var if you don't want Cal to collect anonymous usage
-NEXT_PUBLIC_TELEMETRY_KEY=js.2pvs2bbpqq1zxna97wcml.oi2jzirnbj1ev4tc57c5r
```
Did the key changed? Or this this unintentional?

vklimontovich replied:
Yes, it is intentional. Since all events are sent from server-side, there's no need to expose the key to client rendering

zomars replied:
Sorry @vklimontovich I0ve meant the key VALUE itself changed. Was it regenerated?

vklimontovich replied:
No, it's the same API, but different key (used for server-to-server communication with Jitsu)

zomars replied:
Gotcha!

zomars commented on apps/web/pages/_app.tsx:
```diff
@@ -30,23 +30,20 @@ function MyApp(props: AppProps) {
-          <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
-        </Head>
-        {Component.requiresLicense ? (
-          <LicenseRequired>
-            <Component {...pageProps} err={err} />
-          </LicenseRequired>
```
Why was this removed?

vklimontovich replied:
Seems like an artifact of merge conflict resolving, it's not intentional

zomars replied:
Let's restore this component and we're good to merge 👍🏽

zomars reviewed (commented):
Code looks great and tidy. Left some Qs to address before merging

hariombalhara reviewed (changes requested):
Thank you for the contribution @vklimontovich !!

I don't see the calls going from the server after adding TELEMETRY_DEBUG=true flag. Enabled proxy as well for terminal and still doesn't see the calls. I am sure I am missing something. Does it work for others ?

Also, left a few comments

vklimontovich committed:
Merge branch 'calcom:main' into main

vklimontovich committed:
Switched to stable version of next-collect; restored LicenseProvider accidentally deleted during merge

vklimontovich committed:
Switched to stable version of next-collect; restored LicenseProvider accidentally deleted during merge

vklimontovich committed:
Updated to latest version of next-collect

vklimontovich committed:
- Updated to latest version of next-collect
- Few improvements in event collection: isTeamBooking for all events, page_url for all events
- Do not send second page event on re-render of /team/[slug]

vklimontovich committed:
Merge branch 'main' of https://github.com/calcom/cal.com into main

# Conflicts:
#	yarn.lock

vklimontovich commented on apps/web/lib/telemetry.ts:
```diff
@@ -1,17 +1,10 @@
- */
 export const telemetryEventTypes = {
   pageView: "page_view",
+  apiCall: "api_call",
```
Adding this here won't fire event on each API call. See nextCollectBasicSettings => eventTypes => `{ "/api*": null }` - automated tracking of api_call events is disabled. You could delete it, but I'd keep it for future use to keep nomenclature consistent

vklimontovich commented on apps/web/pages/team/[slug].tsx:
```diff
@@ -34,15 +34,12 @@ function TeamPage({ team }: TeamPageProps) {
-          isTeamBooking: true,
-        })
-      )
+    telemetry.event(
```
It could. I kept it for compatibility. However, I don't think duplicated events is a problem. Do you could sum(page_views) as opposed to distinct page views?

vklimontovich commented on apps/web/components/booking/pages/BookingPage.tsx:
```diff
@@ -332,12 +333,12 @@ const BookingPage = ({
   const bookEvent = (booking: BookingFormValues) => {
-    telemetry.withJitsu((jitsu) =>
-      jitsu.track(
-        top !== window ? telemetryEventTypes.embedBookingConfirmed : telemetryEventTypes.bookingConfirmed,
```
You're right, will fix it

vklimontovich committed:
Revert booking confirmed tracking

vklimontovich committed:
Merge branch 'calcom:main' into main

vklimontovich committed:
Applied prettier + fix lint

vklimontovich committed:
Merge remote-tracking branch 'origin/main' into main
